vignettes/MetaboCoreUtils.Rmd

```diff
@@ -224,6 +224,322 @@ rtime$rindex_cor <- correctRindex(rtime$rindex_r, ref)
+explain the measured abundances (as in our example above).
+
+```{r}
+qc_lm[qc_lm_summary[, "p.value"] > 0.05] <- NA
+```
+
+We can next adjust the data for the estimated signal drifts using the
+`adjust_lm` function. We will thus adjust abundances in all samples (including
+the study samples) using the linear models estimated on the QC samples. For
+features for which no linear model is provided (i.e., with an `NA` in the `list`
+of linear models) the original abundances will be returned *as is*. With
+parameter `data` we need to provide a `data.frame` with all required covariates
+for the fitted models (i.e., defined by the `formula` passed to the `fit_lm`
+call). Also, since we fitted the models to the data in `log2` scale, we need
+also to provide log2 transformed values to the `adjust_lm` function.
+
+```{r}
+#' Adjust the data for the estimated signal drift
+vals_adj <- adjust_lm(log2(vals), data = sdata, lm = qc_lm)
+
+#' Transform data again into natural scale
+vals_adj <- 2^vals_adj
+```
+
+Finally, we can (and should) evaluate the impact of the adjustment by plotting
+the raw and adjusted values into the same plot. Below we plot these values (raw
+values as open circles, adjusted values as filled circles) for the 2nd feature.
+
+```{r, fig.cap = "Feature abundances before (open circles) and after adjustment (filled circles). The grey dashed line represents the injection index dependent signal drift estimated on the raw data and the solid grey line the same model estimated on the adjusted data."}
+plot(x = sdata$injection_index, y = log2(vals[2, ]),
+     xlab = "injection_index", ylab = expression(log[2]~abundance),
+     col = "#00000080")
+points(x = sdata$injection_index, y = log2(vals_adj[2, ]),
+       pch = 16, col = "#00000080")
+grid()
+abline(qc_lm[[2]], col = "grey", lty = 2)
+
+#' fit a model to the QC samples of the adjusted data
+l <- lm(log2(vals_adj[2, qc_index]) ~ sdata$injection_index[qc_index])
+abline(l, col = "grey")
+
+```
+
+As expected, the signal drift was removed by the adjustment.
+
+We can also evaluate the performance of the whole adjustment by comparing the
+correlation of abundances with injection index before and after
+adjustment. Below we calculate the correlation between abundances in QC samples
+and the respective injection index of these samples using the non-parametric
+Spearman method.
+
+
+We restrict the calculation to features that were also adjusted
+using the signal dependent
+
+```{r}
+#' Identify features for which the adjustment was performed
+fts_adj <- which(!is.na(qc_lm))
+
+#' Define a function to calculate the correlation
+cor_fun <- function(i, y) {
+    values <- y[i, qc_index]
+    if (sum(!is.na(values)) >= 9)
+        cor(values, sdata$injection_index[qc_index],
+            method = "spearman", use = "pairwise.complete.obs")
+    else NA_real_
+}
+
+#' Calculate correlations for raw data, skipping features for
```

Comment:
little typo here "skipping *for features*..."

Reply:
fixed.